### PR TITLE
[rllib] Propagate model options correctly in ARS / ES, to action dist of PPO

### PR DESCRIPTION
**Pull request by ericl**
cc @eugenevinitsky

### CODE REVIEW

File: python/ray/rllib/agents/ars/policies.py [hunk 1]
```diff
@@ -110,27 +100,16 @@ def get_weights(self):
...
-        GenericPolicy.__init__(
-            self,
-            sess,
-            action_space,
-            preprocessor,
-            observation_filter,
-            action_noise_std,
-            options=options)
+                 model_options, action_noise_std):
+        GenericPolicy.__init__(self, sess, action_space, preprocessor,
```
[the remark this answers is not in the document]
Comment (ericl) [reply python/ray/rllib/agents/ars/policies.py]: Hmm I think that should work. I rewrote it to use the standard model option instead of explicitly having a linear policy.

File: python/ray/rllib/models/catalog.py [hunk 2]
```diff
@@ -51,14 +51,15 @@ class ModelCatalog(object):
         >>> prep = ModelCatalog.get_preprocessor(env)
         >>> observation = prep.transform(raw_observation)
 
-        >>> dist_cls, dist_dim = ModelCatalog.get_action_dist(env.action_space)
-        >>> model = ModelCatalog.get_model(inputs, dist_dim)
+        >>> dist_cls, dist_dim = ModelCatalog.get_action_dist(
```
Comment (richardliaw) [inline python/ray/rllib/models/catalog.py]: why these changes (making config/option mandatory)?

File: python/ray/rllib/examples/carla/scenarios.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
+"""Collection of Carla scenarios, including those from the CoRL 2017 paper."""
```
Comment (richardliaw) [inline python/ray/rllib/examples/carla/scenarios.py]: do you need `future` import here?

File: python/ray/rllib/examples/carla/train_ppo.py [hunk 4]
```diff
@@ -28,14 +28,14 @@
     "carla": {
         "run": "PPO",
         "env": "carla_env",
-        "trial_resources": {"cpu": 4, "gpu": 1},
```
Comment (richardliaw) [inline python/ray/rllib/examples/carla/train_ppo.py]: why does Carla no longer need GPU specification?

File: python/ray/rllib/models/catalog.py (hunk 2 above)
Comment (ericl) [reply python/ray/rllib/models/catalog.py]: This prevents you from forgetting to specify them. We should have made it mandatory in the first place.

File: python/ray/rllib/examples/carla/train_ppo.py (hunk 4 above)
Comment (ericl) [reply python/ray/rllib/examples/carla/train_ppo.py]: It's auto-detected for rllib since 0.5.0.

File: python/ray/rllib/examples/carla/scenarios.py (hunk 3 above)
Comment (ericl) [reply python/ray/rllib/examples/carla/scenarios.py]: These files are just moved verbatim.

**Review by richardliaw — APPROVED** [review]
